### ISSUE TITLE
Circular dependencies giving rollup fits

### ISSUE THREAD

**Issue by GreenGremlin** [cheeriojs/cheerio#1311]
Circular dependencies are valid in cjs, but create all kinds of issues when converted to es6. I'm not familiar with the Cheerio code, but wanted to open this issue to see how feasible it may be to factor out Cheerio's circular dependencies.

### Rollup output (slightly re-formatted)

```
Circular dependency: node_modules/cheerio/lib/cheerio.js ->
        node_modules/cheerio/lib/api/attributes.js ->
        node_modules/cheerio/lib/static.js ->
        node_modules/cheerio/lib/cheerio.js
Circular dependency: node_modules/cheerio/lib/cheerio.js ->
        node_modules/cheerio/lib/api/attributes.js ->
        node_modules/cheerio/lib/static.js ->
        commonjs-proxy-/Users/jonathan/my-package/node_modules/cheerio/lib/cheerio.js ->
        node_modules/cheerio/lib/cheerio.js
```

**Comment by fb55** [comment]
Happy to accept PRs :)

**Comment by Sureiya** [comment]
This doesn't exist in 1.0.0 branch but does exist in 1.0.0rc.3,  there is another circular dependency happening with html2parser versions less than 4.0.0 as well.

**Comment by tobiasmuehl** [comment]
Any updates on this?

**Comment by fb55** [comment]
Just published a new release, which should resolve this if it is resolved on the master branch.

**Comment by tvanc** [comment]
I'm still getting this exact issue on 1.0.0-rc5

**Comment by 5saviahv** [comment]
It is not fixed in 1.0.0-rc5 yet

**Comment by fb55** [comment]
Fixed in RC6.
